source/ota.c

```diff
@@ -86,182 +95,386 @@ typedef struct OtaStateTableEntry
+ */
 static OtaErr_t setImageStateWithReason( OtaImageState_t stateToSet,
                                          uint32_t reasonToSet );
 
-/* A helper function to cleanup resources during OTA agent shutdown. */
+/**
+ * @brief Internal function to update the job status to the jobs service from current image state.
+ *
+ * @param state State to set.
+ * @param subReason Reason for status.
+ * @return OtaErr_t OtaErrNone if successful, other codes on failure.
+ */
+static OtaErr_t updateJobStatusFromImageState( OtaImageState_t state,
+                                               int32_t subReason );
 
+/**
+ * @brief  A helper function to cleanup resources during OTA agent shutdown.
+ */
 static void agentShutdownCleanup( void );
 
-/* A helper function to cleanup resources when data ingestion is complete. */
-
+/**
+ * @brief A helper function to cleanup resources when data ingestion is complete.
+ */
 static void dataHandlerCleanup( IngestResult_t result );
 
-/*
- * Prepare the document model for use by sanity checking the initialization parameters
- * and detecting all required parameters.
+/**
+ * @brief Prepare the document model for use by sanity checking the initialization parameters and detecting all required parameters.
+ *
+ * @param pDocModel Details of expected parameters in the job doc.
+ * @param pBodyDef Structure to store the details of keys and where to store them.
+ * @param contextBaseAddr Start of file context.
+ * @param contextSize Size of file context.
+ * @param numJobParams Number of parameters to be extracted.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
  */
-
 static DocParseErr_t initDocModel( JsonDocModel_t * pDocModel,
                                    const JsonDocParam_t * pBodyDef,
                                    void * contextBaseAddr,
                                    uint32_t contextSize,
                                    uint16_t numJobParams );
 
-/* Check if the platform is in self-test. */
+/**
+ * @brief Initialize buffers for storing the file attributes.
+ *
+ * @param pOtaBuffer OTA Application buffers.
+ */
+static void initializeAppBuffers( OtaAppBuffer_t * pOtaBuffer );
 
-static bool inSelftest( void );
+/**
+ * @brief Initialize jobId and protocol buffers.
+ */
+static void initializeLocalBuffers( void );
+
+/**
+ * @brief Search the state transition table for the entry based on current state and incoming event.
+ *
+ * @param pEventMsg Incoming event information.
+ * @return uint32_t Index of the transition.
+ */
+static uint32_t searchTransition( const OtaEventMsg_t * pEventMsg );
+
+/**
+ * @brief Initiate download if not in self-test else reboot
+ *
+ * @return OtaErr_t OtaErrNone if successful.
+ */
+static OtaErr_t processValidFileContext( void );
+
+/**
+ * @brief Validate update version when receiving job doc in self test state.
+ *
+ * @param pFileContext Stores file information
```

Comment:
Missed a period :)

Reply:
Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/116/commits/3e5b1d96211d212a79fd5aa4a1bd33a51add1e85